Makefile

```diff
@@ -394,14 +394,16 @@ gen: \
 	coderd/database/querier.go \
 	provisionersdk/proto/provisioner.pb.go \
 	provisionerd/proto/provisionerd.pb.go \
-	site/src/api/typesGenerated.ts
+	site/src/api/typesGenerated.ts \
+	cli/testdata/golden
```

Comment:
Updating tests automatically as a result of `make gen` seems like it will cause some changes to get unnoticed. If someone changed the CLI output or an env variable name accidentally, then ran `make gen` before `make test`, then the tests would pass. Then the only line of defense is the code reviewer noticing the mistake and pointing it out.

I like it being a make task but I don't think we should bundle it under `make gen` to avoid this problem. We can put comment in the test explaining what to run to update the golden files.

Reply:
We also probably won't be changing these command outputs very often so I think having to run a different command than usual to get it to work is acceptable. People will see failing tests, find the test and see the comment saying they need to regenerate it.

Reply:
Oh and to prevent people from forgetting to change it we should run it as part of the gen job in CI.

Reply:
Makes sense, great suggestion. What do you think about keeping `.PHONY` in this case so that the command always produces fresh output (e.g. when new test cases are added)?

Reply:
If you add the dependency on all Go files like above then it should work the way you want without being PHONY I believe.